apps/OboeTester/app/src/main/java/com/mobileer/oboetester/TestDataPathsActivity.java

```diff
@@ -267,16 +280,25 @@ public String didTestFail() {
         StreamConfiguration requestedOutConfig = mAudioOutTester.requestedConfiguration;
         StreamConfiguration actualInConfig = mAudioInputTester.actualConfiguration;
         StreamConfiguration actualOutConfig = mAudioOutTester.actualConfiguration;
-        boolean passed = true;
         if (mMaxMagnitude <= MIN_REQUIRED_MAGNITUDE) {
             why += ", mag";
         }
-        if (mPhaseJitter > MAX_ALLOWED_JITTER) {
-            why += ", jitter";
+        if (!isPhaseJitterValid()) {
+            why += ", jitterUnknown";
+        } else if (getAveragePhaseError() > MAX_ALLOWED_JITTER) {
```

Comment:
This statement is using getAveragePhaseError() > MAX_ALLOWED_JITTER while is finished early is using (getAveragePhaseError() < MAX_ALLOWED_JITTER). One of these be should <= or >= so the statements are opposite of one another.

Reply:
The statements are not opposite of each other. One asks whether we fail. The other code asks whether we can finish early because the results look so good.